sklearn/utils/estimator_checks.py

```diff
@@ -523,7 +523,7 @@ def check_estimator_sparse_data(name, estimator_orig):
                           "sparse input is not supported if this is not"
                           " the case." % name)
                     raise
-        except Exception as e:
```

Comment:
Shouldn't it print the exception instead of ignoring it? It kinda feels odd to catch `Exception` and suppress it completely.

Reply:
There's a `raise` statement after the `print` statement below, so the exception will still be visible. It's still strange semantics that the exception does not wrap the `print` message though.

Reply:
There's a `raise`, sure. But it says that the estimator is not failing properly on sparse data, i.e. the message is being very specific, whereas it's catching `Exception`. So if the estimator fails because there's not enough memory, or because there's something wrong with the disk and the estimator reads from the disk while fitting, for whatever reason, we still show the same error message. It's just a bad idea to catch `Exception`, and pretend it's much more specific than `Exception`.

Reply:
Agreed that it's bad that `except Exception` can be catching anything (minus a `TypeError` or  `ValueError` from above) while a specific `print` message is being shown (should probably be a warning).

I am not too familiar with this part of the codebase, but the scope of this PR (just removing unused variables) does not change the prior behavior. This block can be improved in a future PR.

```
In [5]: try:
   ...:     1 / 0
   ...: except Exception as e:
   ...:     print('Something went wrong')
   ...:     raise
   ...:
   ...:
Something went wrong
---------------------------------------------------------------------------
ZeroDivisionError                         Traceback (most recent call last)
<ipython-input-5-b91504a88660> in <module>()
      1 try:
----> 2     1 / 0
      3 except Exception as e:
      4     print('Something went wrong')
      5     raise

ZeroDivisionError: division by zero

In [6]: try:
   ...:     1 / 0
   ...: except Exception:
   ...:     print('Something went wrong')
   ...:     raise
   ...:
   ...:
Something went wrong
---------------------------------------------------------------------------
ZeroDivisionError                         Traceback (most recent call last)
<ipython-input-6-ee8d63f9937e> in <module>()
      1 try:
----> 2     1 / 0
      3 except Exception:
      4     print('Something went wrong')
      5     raise

ZeroDivisionError: division by zero
```